apps/anoma_lib/lib/noun/jam.ex

```diff
@@ -357,6 +363,13 @@ defmodule Noun.Jam do
     # now we have the actual length and can read that many bits off the
     # bitstream. shadowing bits and length once more.
     size = size - length
+    IO.puts("size: #{size}, length: #{length}")
+
+    if size < 0 do
+      raise CueError,
+            "invalid atom length at offset #{inspect(offset)}"
+    end
+
```

Comment:
I fail to see how this protects anything, matching on the left hand side with a negative number would raise an error